### PR TITLE
FM2-565: Add support for searching on FulfillerStatus

### PR DESCRIPTION
**Pull request by mogoodrich**
## Issue I worked on

see https://issues.openmrs.org/browse/FM2-

## Checklist: I completed these to help reviewers :)

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by mogoodrich** [comment]
Note that test will fail until [TRUNK-](https://issues.openmrs.org/browse/TRUNK-6176) is merged in, so this PR is dependent on that ticket.

File: api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirMedicationRequestDaoImpl.java
Comment (mogoodrich) [inline api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirMedicationRequestDaoImpl.java:115]: Typo in previous commit I discovered

File: api/src/main/java/org/openmrs/module/fhir2/providers/r3/EncounterFhirResourceProvider.java
Comment (mogoodrich) [inline api/src/main/java/org/openmrs/module/fhir2/providers/r3/EncounterFhirResourceProvider.java:199]: this query has been tweaked to include "declined" medication requests as ones that are excluded when determining "active"; this requires the new fulfiller status enum values defined in TRUNK-6176

File: integration-tests-2.7/src/test/java/org/openmrs/module/fhir2/provider/r3/EncounterFhirResourceProvider_2_7IntegrationTest.java
Comment (mogoodrich) [inline integration-tests-2.7/src/test/java/org/openmrs/module/fhir2/provider/r3/EncounterFhirResourceProvider_2_7IntegrationTest.java:39]: I had to move this to a new 2.7 package because tihs test will require the new "DECLINED" order fulfiller status

File: api/src/main/java/org/openmrs/module/fhir2/providers/r3/EncounterFhirResourceProvider.java
Comment (mseaton) [inline api/src/main/java/org/openmrs/module/fhir2/providers/r3/EncounterFhirResourceProvider.java:165]: Is this comment true?  This method will still work, right?  It just won't exclude declined, but that's ok, because nothing will _be_ declined in pre-2.6.1.

File: api/src/main/java/org/openmrs/module/fhir2/providers/r3/EncounterFhirResourceProvider.java
Comment (mseaton) [reply api/src/main/java/org/openmrs/module/fhir2/providers/r3/EncounterFhirResourceProvider.java:199]: Defer to @ibacher and you @mogoodrich as to whether this "active" indication is something that should be changed like this, but seems fine to me

**Review by mseaton — APPROVED** [review]
All looks generally good to me, though would want @ibacher to have a look

File: api/src/main/java/org/openmrs/module/fhir2/providers/r3/EncounterFhirResourceProvider.java
Comment (mogoodrich) [reply api/src/main/java/org/openmrs/module/fhir2/providers/r3/EncounterFhirResourceProvider.java:165]: Unfortunately, it doesn't work because the underlying Hibernate search needs to map declined to the enum, and it fails when declined doesn't exist.  This is why I needed to move the test for this method to the new 2.7 integration submodule.

File: api/src/main/java/org/openmrs/module/fhir2/providers/r3/EncounterFhirResourceProvider.java
Comment (mogoodrich) [reply api/src/main/java/org/openmrs/module/fhir2/providers/r3/EncounterFhirResourceProvider.java:199]: This search endpoint has been designed as a custom endpoint for the Dispensing app, so it really needs to do whatever business logic the Dispense app requires... that being said, it's a good point that if that is the intent, the endpoint should be be commented and/or named better to make that clear... I will take a look next week.

**Comment by mogoodrich** [comment]
@ibacher @mseaton  updated the javadocs on the "encountersWithMedicationRequests" query to note that it has been designed to work with the Dispense ESM and that the contract may change.

@ibacher  LGTM if this works for you (and sorry for the bombardment of PRs to review... PATCH up next, but, don't worry, that won't be ready until tomorrow at the earliest!)

File: api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseEncounterDao.java
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseEncounterDao.java:167]: My only real suggestion here is that maybe these should be non-abstract and just do nothing? That way, they get implemented where we care about them (basically FhirEncounterDaoImpl_2_2), but can be skipped in other classes where the implementation don't make sense?

**Review by ibacher — APPROVED** [review]
Thanks @mogoodrich. One small question, but this looks fine to me.

File: api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseEncounterDao.java
Comment (mogoodrich) [reply api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseEncounterDao.java:167]: Make sense, will update...

**Comment by mogoodrich** [comment]
@ibacher  I made the change you suggested re: the abstract methods, thanks, I'm going to go ahead and merge!

**Pull request merged**
